homeassistant/components/speedtestdotnet/sensor.py

```diff
@@ -70,46 +82,24 @@ def should_poll(self):
     def device_state_attributes(self):
         """Return the state attributes."""
         attributes = {ATTR_ATTRIBUTION: ATTRIBUTION}
-        if self._data is not None:
-            return attributes.update(
-                {
-                    ATTR_BYTES_RECEIVED: self._data["bytes_received"],
-                    ATTR_BYTES_SENT: self._data["bytes_sent"],
-                    ATTR_SERVER_COUNTRY: self._data["server"]["country"],
-                    ATTR_SERVER_ID: self._data["server"]["id"],
-                    ATTR_SERVER_LATENCY: self._data["server"]["latency"],
-                    ATTR_SERVER_NAME: self._data["server"]["name"],
-                }
+        if self.type == "download":
+            attributes.update(
+                {ATTR_BYTES_RECEIVED: self.coordinator.data["bytes_received"]}
             )
+        if self.type == "upload":
+            attributes.update({ATTR_BYTES_SENT: self.coordinator.data["bytes_sent"]})
+        attributes.update(
+            {
+                ATTR_SERVER_NAME: self.coordinator.data["server"]["name"],
+                ATTR_SERVER_COUNTRY: self.coordinator.data["server"]["country"],
+                ATTR_SERVER_ID: self.coordinator.data["server"]["id"],
+            }
+        )
         return attributes
 
     async def async_added_to_hass(self):
         """Handle entity which will be added."""
-        await super().async_added_to_hass()
-        state = await self.async_get_last_state()
-        if not state:
-            return
-        self._state = state.state
 
         self.async_on_remove(
-            async_dispatcher_connect(
-                self.hass, DATA_UPDATED, self._schedule_immediate_update
-            )
+            self.coordinator.async_add_listener(self.async_write_ha_state)
         )
-
-    def update(self):
```

Comment:
Please implement an `async_update` method which awaits `coordinator.async_request_refresh`. Then we allow users to manually update an entity using the general entity update service.